packages/xarc-app/src/index.ts

```diff
@@ -0,0 +1,207 @@
+         * when trying to load CSS modules.
+         *
+         * Set this to false to avoid loading the ignore hook.
+         */
+        ignoreStyles?: boolean;
+        /**
+         * Set to false to avoid loading node.js require hook to handle isomorphic assets
+         */
+        isomorphicExtendRequire?: boolean;
+      }
+    | Function,
+  callback?: Function
+) {
+  if (typeof options === "function") {
+    callback = options;
+    options = {};
+  } else {
+    options = options || {};
+  }
+
+  if (options.babelRegister) {
+    const babelRegister = optionalRequire("@babel/register");
+    if (!babelRegister) {
+      console.log(
+        "To use @babel/register mode, you need to install the @babel/register and support modules."
+      );
+      console.log("Please see documentation for more details.");
+      return process.exit(1);
+    }
+    const regOptions = Object.assign(
+      {
+        extensions: [".es6", ".es", ".jsx", ".js"]
+      },
+      options.babelRegister || {}
+    );
+
+    logger.info(
+      `Loading @babel/register for runtime transpilation files (extensions: ${regOptions.extensions}).`
+    );
+    logger.info(`The transpilation only occurs the first time you load a file.`);
+    babelRegister(regOptions);
+  }
+
+  /**
+   * css-modules-require-hook: handle css-modules on node.js server.
+   * similar to Babel's babel/register it compiles CSS modules in runtime.
+   *
+   * generateScopedName - Short alias for the postcss-modules-scope plugin's option.
+   * Helps you to specify the custom way to build generic names for the class selectors.
+   * You may also use a string pattern similar to the webpack's css-loader.
+   *
+   * https://github.com/css-modules/css-modules-require-hook#generatescopedname-function
+   * https://github.com/webpack/css-loader#local-scope
+   * https://github.com/css-modules/postcss-modules-scope
+   */
+  if (options.cssModuleHook === true) {
+    const opts = Object.assign(
+      {
+        generateScopedName: "[name]__[local]___[hash:base64:5]",
+        extensions: [".scss", ".styl", ".less", ".css"],
+        preprocessCss: function(css, filename) {
+          if (filename.endsWith(".styl")) {
+            return require("stylus")(css)
+              .set("filename", filename)
+              .render();
+          } else if (filename.endsWith(".scss")) {
+            return require("node-sass").renderSync({
+              css,
+              file: filename
+            }).css;
+          } else {
+            return css;
+          }
+        },
+        processorOpts: { parser: require("postcss-less").parse }
+      },
+      options.cssModuleHook || {}
+    );
+
+    cssModuleHook(opts);
+  } else if (options.ignoreStyles !== false) {
+    optionalRequire("ignore-styles");
+  }
+
+  if (options.isomorphicExtendRequire !== false) {
+    return isomorphicExtendRequire();
+  }
+
+  return callback ? callback() : Promise.resolve();
```

Comment:
hmmm, my last comment didn't go through on this line, but i answer my question above :).

perhaps instead of `Function` we use `() => Promise<void>` for the type for `callback`?

Reply:
yup, definitely will be reiterating to improve types.